src/dune_pkg/opam_solver.ml

```diff
@@ -750,6 +862,23 @@ module Solver_result = struct
     }
 end
 
+(* Returns the latest version of the package named
+   "ocaml-base-compiler" which does not have the avoid-version flag
+   set. This will be used to allow the solver to prefer non-development
+   versions of compiler packages, working around the lack of support for
+   the avoid-version flag in opam-0install-solver. *)
+let latest_released_base_compiler_version repos =
+  let+ all_versions =
+    OpamPackage.Name.of_string "ocaml-base-compiler" |> Opam_repo.load_all_versions repos
```

Comment:
Seems a bit annoying to do this work if `version_preference = Oldest`. Maybe just set this to None in that case?

Reply:
done